cmd/crates/soroban-rpc/src/txn.rs

```diff
@@ -315,13 +315,17 @@ pub fn assemble(
 
     // update the fees of the actual transaction to meet the minimum resource fees.
     let classic_transaction_fees = crate::DEFAULT_TRANSACTION_FEES;
-    // Pad the fees up by 15% for a bit of wiggle room.
-    tx.fee = (tx.fee.max(
-        classic_transaction_fees
-            + u32::try_from(simulation.min_resource_fee)
-                .map_err(|_| Error::LargeFee(simulation.min_resource_fee))?,
-    ) * 115)
+
+    // Pad the fees up by 15% for a bit of wiggle room
+    let padded_fee = u64::from(
```

Comment:
I don't think this logic is correct. We should add 15% on top of the resource fee, not the combined fee (or even more specifically to the refundable portion of the resource fee, as non-refundable fee has almost no wiggle room).
It's also weird that this computation adds the inclusion fee at all; that should probably rather happen on the client side, or at least the inclusion fee should be passed as a part of simulation request. But that's a separate issue.

Reply:
Agreed, but this change is addressing the overflow issue, and we should keep it narrowly focused on that.

Reply:
Yeah, that's true for the second part of my comment. But I still think it would be less confusing if we adjusted the resource fee, then added classic fee (at the very least it would make the code easier to read).

Reply:
This is the client side - you can think of this code as the Rust RPC client.

> We should add 15% on top of the resource fee, not the combined fee (or even more specifically to the refundable portion of the resource fee

Sure, no argument here - I'm trying to maintain the original logic and only add the overflow checks to minimize the behavior check. Totally fine with tweaking the algorithm itself but we should do that separately.

Reply:
> This is the client side - you can think of this code as the Rust RPC client.
FWIW I referred here specifically to the interface exposed to the user. I find it surprising that it's currently not possible to adjust the inclusion fee.
Anyway, I don't think this logic is set in stone, but I'm also fine with leaving it as is for now, given that this is going to be changed soon.

Reply:
> it's currently not possible to adjust the inclusion fee

you can, since it takes the max of the existing vs. estimated fees. but agreed that it's moot anyway.

Reply:
What I mean is to explicitly set an inclusion fee for transaction. IIUC CLI only allows overriding the whole fee. If we also do max of user-provided fee and resource fee, then we're confusing the user (they might say they want to spend 1000 stroops, while simulation ends up with 10 XLM).

Reply:
yep good point, they should be separate overrides :+1: